### PR TITLE
Dtspo 17156 finops dashboard

### PR DESCRIPTION
**Pull request by nman-hmcts**
### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: components/law-finops-dashboard/init.tf [hunk 1]
```diff
@@ -0,0 +1,17 @@
+terraform {
+  required_version = ">= 1.0.10"
+
+  backend "azurerm" {
+  }
+
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "3.10.0"
```
Comment (thomast1906) [inline components/law-finops-dashboard/init.tf:10]: Why a low tf version?

**Review by thomast1906 — CHANGES_REQUESTED** [review]
Worth having `30d` as a parameter value ?

File: components/law-finops-dashboard/dash.tpl [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+                {
+                  "name": "DraftRequestParameters",
+                  "isOptional": true
+                },
+                {
+                  "name": "Query",
+                  "value": "find where TimeGenerated between(startofday(ago(1d)) .. startofday(now())) project _ResourceId, _BilledSize, _IsBillable, ResourceType\n| where _IsBillable == true \n| summarize BillableDataBytes = sum(_BilledSize) by _ResourceId\n| extend ResourceName = tostring(split(_ResourceId, \"/\")[8])\n| extend ResourceProvider = tostring(split(_ResourceId, \"/\")[6])\n| extend ResourceType = tostring(split(_ResourceId, \"/\")[7])\n| extend ResourceGroup = tostring(split(_ResourceId, \"/\")[4])\n| summarize BillableDataBytes = sum(BillableDataBytes) by ResourceName, ResourceProvider, ResourceType, ResourceGroup\n| extend DataIngestedInGB = round(BillableDataBytes / 1024 / 1024 / 1024, 2)\n| extend DataIngested = strcat(round(DataIngestedInGB, 2), \" GB\")\n| extend EstimatedDataIngestionCosts = strcat(\"£\", round(DataIngestedInGB * 2.27, 2))\n| project-away BillableDataBytes, DataIngestedInGB\n| sort by DataIngested nulls last\n",
+                  "isOptional": true
```
[the remark this answers is not in the document]
Comment (nman-hmcts) [reply components/law-finops-dashboard/dash.tpl:157]: good spot @thomast1906 this needs to be updated with the variable of cost per gb too. I have just done it.

File: components/law-finops-dashboard/variables.tf [hunk 3]
```diff
@@ -0,0 +1 @@
+../../environments/variables.tf
```
Comment (thomast1906) [inline components/law-finops-dashboard/variables.tf:1]: is that symlink correctly?

Comment (nman-hmcts) [reply components/law-finops-dashboard/variables.tf:1]: I beleive it does, or else this dashboard would have never deployed. https://portal.azure.com/#@hmcts.net/dashboard/arm/subscriptions/7a4e3bd5-ae3a-4d0c-b441-2188fee3ff1c/resourcegroups/law-finops-dashboard-test-rg/providers/microsoft.portal/dashboards/hmcts-qa-law-finops-dashboard

Comment (thomast1906) [reply components/law-finops-dashboard/variables.tf:1]: ty

**Review by thomast1906 — APPROVED** [review]

**Pull request merged**